external_data/task_functions.py

```diff
@@ -0,0 +1,534 @@
+    df.loc[df["colter_code_insee"] == "75056", "colter_niveau"] = "particulier"
+
+    dfcolter = pd.concat([dfcolter, df])
+
+    if os.path.exists(data_dir + "colter-new.csv"):
+        os.remove(data_dir + "colter-new.csv")
+
+    dfcolter.to_csv(data_dir + "colter-new.csv", index=False)
+
+
+def preprocess_convcollective_data(
+    data_dir,
+) -> None:
+    os.makedirs(os.path.dirname(data_dir), exist_ok=True)
+    cc_url = (
+        "https://www.data.gouv.fr/fr/datasets/r/bfc3a658-c054-4ecc-ba4b-22f3f5789dc7"
+    )
+    r = requests.get(cc_url)
+    with open(data_dir + "convcollective-download.csv", "wb") as f:
+        for chunk in r.iter_content(1024):
+            f.write(chunk)
+
+    cc_df = pd.read_csv(
+        data_dir + "convcollective-download.csv",
+        dtype=str,
+        names=["mois", "siret", "idcc", "date_maj"],
+        header=0,
+    )
+    cc_df["siren"] = cc_df["siret"].str[0:9]
+    cc_df["idcc"] = cc_df["idcc"].apply(lambda x: str(x).replace(" ", ""))
+    liste_cc = (
+        cc_df.groupby(by=["siren"])["idcc"].apply(list).reset_index(name="liste_idcc")
+    )
+    liste_cc.to_csv(data_dir + "convcollective-new.csv", index=False)
+
+
+def preprocess_finess_data(
+    data_dir,
+) -> None:
+    os.makedirs(os.path.dirname(data_dir), exist_ok=True)
+    r = requests.get(
+        "https://www.data.gouv.fr/fr/datasets/r/2ce43ade-8d2c-4d1d-81da-ca06c82abc68"
+    )
+    with open(data_dir + "finess-download.csv", "wb") as f:
+        for chunk in r.iter_content(1024):
+            f.write(chunk)
+
+    df = pd.read_csv(
+        data_dir + "finess-download.csv",
+        dtype=str,
+        sep=";",
+        encoding="Latin-1",
+        skiprows=1,
+        header=None,
+    )
+    df = df[[1, 18, 22]]
+    df = df.rename(columns={1: "finess", 18: "cat_etablissement", 22: "siren"})
+    df["siren"] = df["siren"].str[:9]
+    df = df[df["siren"].notna()]
+    res = df.groupby(["siren"])["finess"].apply(list).reset_index(name="liste_finess")
+    res2 = (
+        df.groupby(["siren"])["cat_etablissement"]
+        .apply(list)
+        .reset_index(name="liste_cat_etablissement")
+    )
+    res = res.merge(res2, on="siren", how="left")
+    res = res[["siren", "liste_finess"]]
+    res.to_csv(data_dir + "finess-new.csv", index=False)
+
+
+def process_elus_files(url, colname):
+    df = pd.read_csv(url, dtype=str, sep="\t")
+    df = df[
+        [
+            colname,
+            "Nom de l'élu",
+            "Prénom de l'élu",
+            "Code sexe",
+            "Date de naissance",
+            "Libellé de la fonction",
+        ]
+    ]
+    df = df.rename(
+        columns={
+            colname: "colter_code",
+            "Nom de l'élu": "nom_elu",
+            "Prénom de l'élu": "prenom_elu",
+            "Code sexe": "sexe_elu",
+            "Date de naissance": "date_naissance_elu",
+            "Libellé de la fonction": "fonction_elu",
+        }
+    )
+    return df
+
+
+def preprocess_elu_data(
```

Comment:
Same for all preprocessing functions, I think it's a good idea to ass a little description of what that preprocessing entails in a doctoring for example.